### PR TITLE
Adds more robust checking of the status code during compute engine auth.

### LINKED ISSUE #23: GCE error when --no-scopes used
From google/google-api-ruby-client#207

In compute_service_account.rb we do:

        response = connection.get 'http://metadata/computeMetadata/v1beta1/instance/service-accounts/default/token'

However, if the VM was created with --no-scopes this will return an error page instead of the token, which usually leads to a subsequent exception while trying to parse it.

**example exception**

```
error_class="MultiJson::ParseError" error="lexical error: invalid char in json text.\n                                       <!DOCTYPE html> <html lang=en> \n                     (right here) ------^\n" plugin_id="object:3f9bc1d9bd80"
  2015-03-25 22:43:22 +0000 [warn]: /opt/google-fluentd/embedded/lib/ruby/gems/2.1.0/gems/multi_json-1.11.0/lib/multi_json/adapters/yajl.rb:11:in `parse'
  2015-03-25 22:43:22 +0000 [warn]: /opt/google-fluentd/embedded/lib/ruby/gems/2.1.0/gems/multi_json-1.11.0/lib/multi_json/adapters/yajl.rb:11:in `load'
  2015-03-25 22:43:22 +0000 [warn]: /opt/google-fluentd/embedded/lib/ruby/gems/2.1.0/gems/multi_json-1.11.0/lib/multi_json/adapter.rb:21:in `load'
  2015-03-25 22:43:22 +0000 [warn]: /opt/google-fluentd/embedded/lib/ruby/gems/2.1.0/gems/multi_json-1.11.0/lib/multi_json.rb:119:in `load'
  2015-03-25 22:43:22 +0000 [warn]: /opt/google-fluentd/embedded/lib/ruby/gems/2.1.0/gems/signet-0.5.1/lib/signet/oauth_2.rb:81:in `parse_json_credentials'
  2015-03-25 22:43:22 +0000 [warn]: /opt/google-fluentd/embedded/lib/ruby/gems/2.1.0/gems/google-api-client-0.7.1/lib/google/api_client/auth/compute_service_account.rb:24:in `fetch_access_token'
  2015-03-25 22:43:22 +0000 [warn]: /opt/google-fluentd/embedded/lib/ruby/gems/2.1.0/gems/signet-0.5.1/lib/signet/oauth_2/client.rb:956:in `fetch_access_token!'
```

**curl results from the affected machine**

```
$ curl 'http://metadata/computeMetadata/v1beta1/instance/service-accounts/default/token'
<!DOCTYPE html>
<html lang=en>
  <meta charset=utf-8>
  <meta name=viewport content="initial-scale=1, minimum-scale=1, width=device-width">
  <title>Error 404 (Not Found)!!1</title>
  <style>
    *{margin:0;padding:0}html,code{font:15px/22px arial,sans-serif}html{background:#fff;color:#222;padding:15px}body{margin:7% auto 0;max-width:390px;min-height:180px;padding:30px 0 15px}* > body{background:url(//www.google.com/images/errors/robot.png) 100% 5px no-repeat;padding-right:205px}p{margin:11px 0 22px;overflow:hidden}ins{color:#777;text-decoration:none}a img{border:0}@media screen and (max-width:772px){body{background:none;margin-top:0;max-width:none;padding-right:0}}#logo{background:url(//www.google.com/images/errors/logo_sm_2.png) no-repeat}@media only screen and (min-resolution:192dpi){#logo{background:url(//www.google.com/images/errors/logo_sm_2_hr.png) no-repeat 0% 0%/100% 100%;-moz-border-image:url(//www.google.com/images/errors/logo_sm_2_hr.png) 0}}@media only screen and (-webkit-min-device-pixel-ratio:2){#logo{background:url(//www.google.com/images/errors/logo_sm_2_hr.png) no-repeat;-webkit-background-size:100% 100%}}#logo{display:inline-block;height:55px;width:150px}
  </style>
  <a href=//www.google.com/><span id=logo aria-label=Google></span></a>
  <p><b>404.</b> <ins>That’s an error.</ins>
  <p>The requested URL <code>/computeMetadata/v1beta1/instance/service-accounts/default/token</code> was not found on this server.  <ins>That’s all we know.</ins>
```

### PR DESCRIPTION
**Pull request by tbetbetbe**
Currently the implementation does not verify that a 200 response is received
before attempting to parse the JSON response.

This PR updates the behaviour to fail with an authorization error, similar to
how other auth library implementations.

Fixes #23

### CODE REVIEW

File: spec/googleauth/compute_engine_spec.rb [hunk 1]
```diff
@@ -60,6 +60,42 @@ def make_auth_stubs(opts = {})
...
+        end
+        c = Faraday.new do |b|
+          b.adapter(:test, stubs)
+        end
+        blk = proc { @client.fetch_access_token!(connection: c) }
+        expect(&blk).to raise_error Signet::AuthorizationError
+        stubs.verify_stubbed_calls
+      end
+
+      it 'should fail if metadata the request returns an unexpected codes' do
```
Comment (sqrrrl) [inline spec/googleauth/compute_engine_spec.rb]: Minor grammatical error, maybe meant 'should fail if the metadata request returns an unexpected code'

File: lib/googleauth/compute_engine.rb [hunk 2]
```diff
@@ -78,6 +89,13 @@ def fetch_access_token(options = {})
         c = options[:connection] || Faraday.default_connection
         c.headers = { 'Metadata-Flavor' => 'Google' }
         resp = c.get(COMPUTE_AUTH_TOKEN_URI)
+        if resp.status == 404
```
Comment (sqrrrl) [inline lib/googleauth/compute_engine.rb]: Minor thing, but case might be a little cleaner here:

```ruby
case resp.status
when 200
  Signet::OAuth2.parse_credentials(resp.body, resp.headers['content-type'])
when 404
  fail(Signet::AuthorizationError, NO_METADATA_SERVER_ERROR)
else
  msg = "Unexpected error code #{resp.status}" + UNEXPECTED_ERROR_SUFFIX
  fail(Signet::AuthorizationError, msg)
end
```

**Comment by sqrrrl** [comment]
I take it the end result is still the same -- an error (albeit a more informative one :) will be raised if --no-scope is used?

**Comment by tbetbetbe** [comment]
That's correct.  That's consistent with how [java](https://github.com/google/google-auth-library-java/blob/master/oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java) and [python](https://github.com/google/oauth2client/blob/master/oauth2client/gce.py) handle it.

File: spec/googleauth/compute_engine_spec.rb (hunk 1 above)
Comment (tbetbetbe) [inline spec/googleauth/compute_engine_spec.rb]: Done

File: lib/googleauth/compute_engine.rb (hunk 2 above)
Comment (tbetbetbe) [inline lib/googleauth/compute_engine.rb]: Done

**Pull request merged**